src/backends/gitlab/API.js

```diff
@@ -0,0 +1,191 @@
+      .then(({ namespace }) => (namespace.kind === "group" ? `/groups/${ encodeURIComponent(namespace.full_path) }` : false));
+  }
+
+  isCollaborator(user) {
+    const WRITE_ACCESS = 30;
+    return this.isGroupProject().then((group) => {
+      if (group === false) {
+        return this.request(`${ this.repoURL }/members/${ user.id }`);
+      } else {
+        return this.request(`${ group }/members/${ user.id }`);
+      }
+    })
+    .then(member => (member.access_level >= WRITE_ACCESS))
+    .catch((err) => {
+        // Member does not have any access. We cannot just check for 404,
+        //   because a 404 is also returned if we have the wrong URI,
+        //   just with an "error" key instead of a "message" key.
+        if (err.status === 404 && err.meta.errorValue["message"] === "404 Not found") {
+          return false;
+        } else {
+          // Otherwise, it is actually an API error.
+          throw err;
+        }
+    });
+  }
+
+  requestHeaders(headers = {}) {
+    return {
+      ...headers,
+      ...(this.token ? { Authorization: `Bearer ${ this.token }` } : {}),
+    };
+  }
+
+  urlFor(path, options) {
+    const cacheBuster = `ts=${ new Date().getTime() }`;
+    const encodedParams = options.params
+          ? Object.entries(options.params).map(
+            ([key, val]) => `${ key }=${ encodeURIComponent(val) }`)
+          : [];
+    return `${ this.api_root }${ path }?${ [cacheBuster, ...encodedParams].join("&") }`;
+  }
+
+  request(path, options = {}) {
+    const headers = this.requestHeaders(options.headers || {});
+    const url = this.urlFor(path, options);
+    return fetch(url, { ...options, headers })
+    .then((response) => {
+      const contentType = response.headers.get("Content-Type");
+      if (options.method === "HEAD") {
+        return Promise.all([response]);
+      }
+      if (contentType && contentType.match(/json/)) {
+        return Promise.all([response, response.json()]);
+      }
+      return Promise.all([response, response.text()]);
+    })
+    .catch(err => Promise.reject([err, null]))
+    .then(([response, value]) => (response.ok ? value : Promise.reject([value, response])))
+    .catch(([errorValue, response]) => {
+      const errorMessageProp = (errorValue && errorValue.message) ? errorValue.message : null;
+      const message = errorMessageProp || (isString(errorValue) ? errorValue : "");
+      throw new APIError(message, response && response.status, 'GitLab', { response, errorValue });
+    });
+  }
+  
+  readFile(path, sha, branch = this.branch) {
+    const cache = sha ? LocalForage.getItem(`gh.${ sha }`) : Promise.resolve(null);
+    return cache.then((cached) => {
+      if (cached) { return cached; }
+      
+      // Files must be downloaded from GitLab as base64 due to this bug:
+      //   https://gitlab.com/gitlab-org/gitlab-ce/issues/31470
+      return this.request(`${ this.repoURL }/repository/files/${ encodeURIComponent(path) }`, {
```

Comment:
https://gitlab.com/gitlab-org/gitlab-ce/issues/31470 was fixed, so we should be able to pull the files directly now.